src/components/UserProfile/BlueSquares/BlueSquare.jsx

```diff
@@ -30,7 +31,7 @@ const BlueSquare = ({ blueSquares, handleBlueSquare, isUserAdmin }) => {
         ))}
       </div>
 
-      {isUserAdmin && (
+      {(hasPermission(role, 'editUserProfile') || hasPermission(role, 'assignOnlyBlueSquares'))  && (
```

Comment:
I notice here `editUserProfile` are `admin` and `owner`, `assignOnlyBlueSquares` are `manager` and `mentor`. Do we want to keep this line to only `hasPermission(role, 'assignBlueSquares')` and add it to admin&owner&manager&mentor

Reply:
Mentor and Manager roles can only assign blue squares and not remove or edit them. That's why I make that verification. If it's a Mentor or Manager they are allowed to add blue squares, if it's an owner or admin they are allowed to add/delete/update blue squares